src/build.sh

```diff
@@ -124,6 +128,41 @@ function build_kernel_main()
     return "$?"
   fi
 
+  if [[ -n "$from_sha_arg" ]]; then
+    # Check if there is a rebase in process.
+    if [[ -d .git/rebase-merge ]]; then
+      printf '%s\n' 'ERROR: Abort the repository rebase before continuing with build from sha (use "git rebase --abort")!'
+      return 22 #EINVAL
+    elif [[ -f .git/MERGE_HEAD ]]; then
+      printf '%s\n' 'ERROR: Abort the repository merge before continuing with build from sha (use "git rebase --abort")!'
+      return 22 #EINVAL
+    elif [[ -f .git/BISECT_LOG ]]; then
+      printf '%s\n' 'ERROR: Stop the repository bisect before continuing with build from sha (use "git bisect reset")!'
+      return 22 #EINVAL
+    elif [[ -d .git/rebase-apply ]]; then
+      printf '%s\n' 'ERROR: Abort the repository patch apply before continuing with build from sha (use "git am --abort")!'
+      return 22 #EINVAL
+    fi
+
+    # Check if given SHA represents real commit
+    cmd_manager "SILENT" "git cat-file -e ${from_sha_arg}^{commit} 2> /dev/null"
+    if [[ "$?" != 0 ]]; then
+      printf '%s\n' 'ERROR: The given SHA does not represent a valid commit sha.'
```

Comment:
Maybe we can output the value of `from_sha_arg` in this print to assert the user about the value being used.